### PR TITLE
Edited error message

### LINKED ISSUE #13115: Error message should tell user how to get to debugging page
#### What happened

`Jetpack.com detected an error. Visit the Jetpack.com debugging page for more information or contact support.`

#### What I expected

`Jetpack.com detected an error. <a href="https://jetpack.com/support/debug/">Visit the Jetpack.com debugging page</a> for more information or contact support.`

Otherwise users have no way of finding the page. It's not in the menu, and doesn't show up in the top search results. The only way I found it was because it happened to be in another GitHub issue that I came across to see if someone had already reported this.

### PR DESCRIPTION
**Pull request by d3vl0p**
Fixes #13115

#### Changes proposed in this Pull Request:

* edited error message to include a link to the debug and contact page

#### Testing instructions:

* This fixes the following issue: https://github.com/Automattic/jetpack/issues/13115

#### Proposed changelog entry for your changes:

* No changelog entry needed

### CODE REVIEW

**Review by jeherve — CHANGES_REQUESTED** [review]
I would recommend linking to the Debug page with the site URL already filled in. I think you could also remove the comment at the end of the line now.

**Comment by omkarbhagat93** [comment]
@jeherve made the change. The debug link will now include `site_url()` at the end of it.

**Review by jeherve — CHANGES_REQUESTED** [review]
The backticks aren't going to work well here. I'd recommend building the string using `sprintf` and `wp_kses` to ensure that translators cannot break the string while translating.

**Comment by omkarbhagat93** [comment]
@jeherve That's great advice. While I knew a little bit about this in theory, I learned so much more in practice while making changes and getting this to work. What do you think about the latest commit?

File: _inc/lib/debugger/class-jetpack-cxn-tests.php [hunk 1]
```diff
@@ -337,10 +337,23 @@ protected function last__wpcom_self_test() {
 
 		remove_filter( 'http_request_timeout', array( 'Jetpack_Debugger', 'jetpack_increase_timeout' ) );
 
+		$error_msg = wp_kses(
+			sprintf(
+				/* translators: 1st placeholder is a link to Jetpack debug page, 2nd one fetches current site URL. */
+				__(
+					'<a href="%1$s%2$s">Visit the Jetpack.com debug page</a> for more information or <a href="https://jetpack.com/contact-support/">contact support</a>.',
```
Comment (mdawaffe) [inline _inc/lib/debugger/class-jetpack-cxn-tests.php]: We should be able to do this with just one placeholder.

File: _inc/lib/debugger/class-jetpack-cxn-tests.php [hunk 2]
```diff
@@ -337,10 +337,23 @@ protected function last__wpcom_self_test() {
...
+					'jetpack'
+				),
+				'https://jetpack.com/support/debug/?url=',
```
Comment (mdawaffe) [inline _inc/lib/debugger/class-jetpack-cxn-tests.php]: If we do just the one place holder, we condense these two into arguments (the debug URL and the site URL) into one:

```php
esc_url( add_query_arg( 'url', urlencode( site_url() ), 'https://jetpack.com/support/debug/' ) )
```

This also ensures we're using the right escaping functions (`esc_url()` for the whole URL, `urlencode()` for the query string parameter) in the right places.

**Review by mdawaffe — DISMISSED** [review]

File: _inc/lib/debugger/class-jetpack-cxn-tests.php (hunk 2 above)
Comment (omkarbhagat93) [reply _inc/lib/debugger/class-jetpack-cxn-tests.php]: Oh! didn't think of that. Also `add_query_arg` is a TIL for me. Thanks. Made the change now.

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good. However, while it works well under Tools > Site Health, we can't really display links in Jetpack > Debug right now:

image


image

You'd need to ensure that the output there allows for links, maybe using `wp_kses` as well:
https://github.com/Automattic/jetpack/blob/fcd387ae443d66632648646772ef0dd676b9bd19/_inc/lib/debugger/class-jetpack-debugger.php#L116

**Comment by omkarbhagat93** [comment]
@jeherve Good point. Thanks for testing. Made the change (also resolved the conflict with the master branch) and this looks good to go!

**Review by jeherve — DISMISSED** [review]
This looks good to me. It should be good to merge. I only wonder if it may be best to have those links open in a new tab, since they direct you out of wp-admin.

**Comment by codestor4** [comment]
@jeherve You're right, opening the link in a new tab makes sense. Let me make that change..

**Comment by omkarbhagat93** [comment]
@jeherve All done. Ready to merge 👍

**Review by jeherve — CHANGES_REQUESTED** [review]
Good addition. I'd have 2 remarks:

- Now that you've added a new attribute to that HTML tag (`target`), you'll want to update the list of attributes you're allowing in `wp_kses`. If you don't, `wp_kses` will strip them for you.
- When you are adding a `target="_blank"` attribute, you also want to add a `rel="noopener noreferrer"` attribute to avoid the problems listed here: https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/ https://stackoverflow.com/a/50709760/1045686

**Comment by codestor4** [comment]
Latest commit includes `noopener noreferrer` and `wp_kses` updated to include `rel` and `target` attributes. Thanks for including links for context. I love that I'm getting to learn so much more by working on a PR :) 

cc: @jeherve

**Review by jeherve — APPROVED** [review]
This looks good to me. Merging.

**Comment by kraftbj** [comment]
One thing I'd want to check is the `wp jetpack status` output of a site in a broken state.

Ideally, in the PR description and testing instructions, there would be enough detail to recreate the error state so it can be tested without figuring out how to break JP.

**Comment by kraftbj** [comment]
2019-10-04 at 2 55 PM

It looks good. I have a vague memory of not including a link for some CLI reason, but that seems to be outdated info.

**Comment by codestor4** [comment]
> Ideally, in the PR description and testing instructions, there would be enough detail to recreate the error state so it can be tested without figuring out how to break JP.

Noted. I'm still learning but I'll remember this for the next time. I wanted to recreate the error message to test things and Jeremy suggested the simplest way is to put it outside of if/else block. That's what I did. I don't know if there is a better way to recreate the error. Can you share some details on how you did it? @kraftbj

**Pull request merged**